ostree-ext/src/container/store.rs

```diff
@@ -1196,6 +1197,37 @@ fn parse_cached_update(meta: &glib::VariantDict) -> Result<Option<CachedImageUpd
     }))
 }
 
+/// Remove any cached
+#[context("Clearing cached update {imgref}")]
+pub fn clear_cached_update(repo: &ostree::Repo, imgref: &ImageReference) -> Result<()> {
+    let cancellable = gio::Cancellable::NONE;
+    let ostree_ref = ref_for_image(imgref)?;
+    let rev = repo.require_rev(&ostree_ref)?;
+    let Some(commitmeta) = repo.read_commit_detached_metadata(&rev, cancellable)? else {
+        return Ok(());
+    };
+
+    // SAFETY: We know this is an a{sv}
```

Comment:
I have no idea what `a{sv}` means here, not a blocker but please enlighten me :smiley:

Reply:
gvariant is a bit like a binary JSON, although also with a richer type system (e.g. 32 and 64 bit signed/unsigned integers, not just "number" etc.).

See https://docs.gtk.org/glib/gvariant-format-strings.html - specifically `a{sv}` is a lot like a default json dictionary (string keys, heterogenous values).

Reply:
thanks for the lesson :mortar_board: